### DISCUSSION TITLE
Roaming Status - AgeOfInformation (or last_updated_timestamp)

### DISCUSSION THREAD

**Discussion by sachinvodafone** [camaraproject/DeviceStatus#98] [General]
In the current DeviceStatus API, it has come to my attention that the CAMARA version lacks an essential piece of information—specifically, the "ageOfInformation" or "last_updated_timestamp" field. This field would play a crucial role in indicating when the roaming status information was last updated.

The absence of this timestamp has led to inconsistencies in the API's behavior across different vendors. While some vendors may provide real-time information, others may rely on historical data. This discrepancy in implementation has resulted in challenges for application developers who seek to ascertain the freshness of the data.

To address this issue, we believe that including a "last_updated_timestamp" or similar field in the API response is essential. This timestamp would empower application developers to make informed decisions based on the recency of the data, regardless of the underlying implementation by different vendors. It would enhance the utility and reliability of the API across various use cases.

**Comment by akoshunyadi** [comment]
I agree that it can be useful. I would prefer a timestamp e.g. lastStatusTime (similar to lastLocationTime)

**Reply by mhfoo** [reply]
Consider to use a HTTP Response Header: Last-Modified

**Reply by eric-murray** [reply]
> Consider to use a HTTP Response Header: Last-Modified

No, the `Last-Modified` header is for an entirely different purpose. That states when the returned record was last changed, not when it was last confirmed to be correct.

The roaming status itself may not change from call to call (at any given time. most end users are not roaming), so the record itself has not been modified, but it may have been checked again to confirm it is still correct. That is the information we want here - when was the roaming status last confirmed, rather than when was the roaming status last modified.

**Reply by mhfoo** [reply]
@eric-murray 
Thanks for providing the context. Agree with your statement.
The attribute name should reflect your explanation.

**Comment by akoshunyadi** [comment]
@sachinvodafone as agreed in the last subproject call, can you please create an issue of this discussion?

**Comment by akoshunyadi** [comment]
The issue #110 has been created, discussion can be closed.
